src/main/scala/com/sky/kafka/configurator/Main.scala

```diff
@@ -28,23 +28,38 @@ object Main extends LazyLogging {
       .text("Session and connection timeout for Zookeeper")
   }
 
-  def main(args: Array[String]): Unit = run(args) match {
-    case Success(_) =>
-      System.exit(0)
-    case Failure(_) =>
-      System.exit(1)
+  def main(args: Array[String]): Unit = {
+    logger.info(s"Running ${BuildInfo.name} ${BuildInfo.version}")
+    run(args) match {
+      case Success(_) =>
+        System.exit(0)
+      case Failure(_) =>
+        System.exit(1)
+    }
   }
 
-  def run(args: Array[String]): Try[Unit] = parse(args) flatMap { conf =>
-    logger.info(s"Running ${BuildInfo.name} ${BuildInfo.version}")
-    val configurator = TopicConfigurator.reader(conf)
-    val result: Try[Unit] = for {
-      topics <- TopicConfigurationParser(new FileReader(conf.file))
-      _ <- topics.map(configurator.configure).sequenceU
+  def run(args: Array[String]): Try[Unit] =
+    parse(args).flatMap { conf =>
+      val configurator = TopicConfigurator.reader(conf)
+      val result = configureTopicsFromFile(conf.file, configurator)
+
+      stopApp(configurator)
+      result
+    }
+
+  private def configureTopicsFromFile(topicConfigYml: File, configurator: TopicConfigurator): Try[Unit] =
+    for {
+      topics <- TopicConfigurationParser(new FileReader(topicConfigYml)).toTry
+      _ <- topics.traverseU { topic =>
+        configurator.configure(topic).run.map {
+          case (logs, _) =>
+            logs.foreach(log => logger.info(log))
+        }.recover {
+          case NonFatal(throwable) =>
+            logger.error(s"Failed to configure ${topic.name}", throwable)
```

Comment:
didn't spot that in my testing, just scrolled up and that was indeed happening. Good spot 😄